fpdf/drawing.py

```diff
@@ -0,0 +1,4207 @@
+
+        Returns:
+            The same tuple as `ImplicitClose.render`.
+        """
+        # pylint: disable=unused-argument
+        rendered, resolved = self.render(gsd_registry, style, last_item)
+        debug_stream.write(f"{self} resolved to {rendered}\n")
+
+        return rendered, resolved
+
+
+class Close(NamedTuple):
+    """
+    A path close element.
+
+    Instructs the renderer to draw a straight line from the end of the last path element
+    to the start of the current path.
+
+    See: `PaintedPath.close`
+    """
+
+    @force_nodocument
+    def render(self, gsd_registry, style, last_item):
+        """
+        Render this path element to its PDF representation.
+
+        Args:
+            gsd_registry (GraphicsStateDictRegistry): the owner's graphics state
+                dictionary registry.
+            style (GraphicsStyle): the current resolved graphics style
+            last_item: the previous path element.
+
+        Returns:
+            a tuple of `(str, new_last_item)`, where `new_last_item` is whatever the old
+            last_item was.
+        """
+        # pylint: disable=unused-argument,no-self-use
+        return "h", last_item
+
+    @force_nodocument
+    def render_debug(self, gsd_registry, style, last_item, debug_stream, pfx):
+        """
+        Render this path element to its PDF representation and produce debug
+        information.
+
+        Args:
+            gsd_registry (GraphicsStateDictRegistry): the owner's graphics state
+                dictionary registry.
+            style (GraphicsStyle): the current resolved graphics style
+            last_item: the previous path element.
+            debug_stream (io.TextIO): the stream to which the debug output should be
+                written. This is not guaranteed to be seekable (e.g. it may be stdout or
+                stderr).
+            pfx (str): the current debug output prefix string (only needed if emitting
+                more than one line).
+
+        Returns:
+            The same tuple as `Close.render`.
+        """
+        # pylint: disable=unused-argument
+        rendered, resolved = self.render(gsd_registry, style, last_item)
+        debug_stream.write(str(self) + "\n")
+
+        return rendered, resolved
+
+
+class DrawingContext:
+    """
+    Base context for a drawing in a PDF
+
+    This context is not stylable and is mainly responsible for transforming path
+    drawing coordinates into user coordinates (i.e. it ensures that the output drawing
+    is correctly scaled).
+    """
+
+    def __init__(self):
+        self._subitems = []
+
+    def add_item(self, item, _copy=True):
+        """
+        Append an item to this drawing context
+
+        Args:
+            item (GraphicsContext, PaintedPath): the item to be appended.
+        """
+
+        if not isinstance(item, (GraphicsContext, PaintedPath)):
+            raise TypeError(f"{item} doesn't belong in a DrawingContext")
+
+        if _copy:
```

Comment:
The quick answer is that this parameter should not be undocumented, so I'll fix that.

The actual technical answer to the question is that performing copies on adding items is default behavior to avoid unexpected results due to the fact that the path objects are not rendered (serialized) until the overall drawing context is rendered. For example,

```python
my_path.style.fill_color = "#00F" # blue
drawing_context.add_item(my_path)

my_path.style.fill_color = "#F00" # red
drawing_context.add_item(my_path)
```

Because `my_path` is reused here, without performing the copy when `add_item` is called, both paths will actually be colored red, rather than the first one being blue and the second one being red as naïvely reading the code would indicate. However, `_copy` is provided as a parameter to the `add_item` method so that advanced use cases can have control over the copy/reference behavior of path reuse.